src/allmydata/storage/server.py

```diff
@@ -708,8 +708,10 @@ def remote_advise_corrupt_share(self, share_type, storage_index, shnum,
         now = time_format.iso_utc(sep="T")
         si_s = si_b2a(storage_index)
         # windows can't handle colons in the filename
-        fn = os.path.join(self.corruption_advisory_dir,
-                          "%s--%s-%d" % (now, str(si_s, "utf-8"), shnum)).replace(":","")
+        fn = os.path.join(
+            self.corruption_advisory_dir,
+            ("%s--%s-%d" % (now, str(si_s, "utf-8"), shnum)).replace(":","")
```

Comment:
This is a (easy to miss) bugfix. Previously the replace was on the _whole_ path, thus removing all colons from paths.